pkg/slo-controller/noderesource/plugins/batchresource/plugin.go

```diff
@@ -113,36 +119,47 @@ func (p *Plugin) Calculate(strategy *configuration.ColocationStrategy, node *cor
 			continue
 		}
 
+		// mark listed pod metric
+		podKey := util.GetPodKey(pod)
+		podMetric, hasMetric := podMetricMap[podKey]
+		podMetricMap[podKey] = nil
+
 		priorityClass := extension.GetPodPriorityClassWithDefault(pod)
 		podRequest := util.GetPodRequest(pod, corev1.ResourceCPU, corev1.ResourceMemory)
 		isPodHighPriority := priorityClass != extension.PriorityBatch && priorityClass != extension.PriorityFree
-		if isPodHighPriority {
-			podHPRequest = quotav1.Add(podHPRequest, podRequest)
-		}
-		podKey := util.GetPodKey(pod)
-		podMetric, ok := podMetricMap[podKey]
-		if !ok {
-			if isPodHighPriority {
-				podHPUsed = quotav1.Add(podHPUsed, podRequest)
-			}
-			podAllUsed = quotav1.Add(podAllUsed, podRequest)
+		if !isPodHighPriority {
 			continue
 		}
 
-		if isPodHighPriority {
+		podHPRequest = quotav1.Add(podHPRequest, podRequest)
+		if hasMetric {
 			podHPUsed = quotav1.Add(podHPUsed, getPodMetricUsage(podMetric))
+		} else {
+			podHPUsed = quotav1.Add(podHPUsed, podRequest)
+		}
+	}
+
+	// When the reported pods is missing in the current pod list, add them into the HP used.
+	hasMissingPod := false
+	for podKey, podMetric := range podMetricMap {
+		if podMetric == nil { // pod is in the current pod list
+			continue
 		}
-		podAllUsed = quotav1.Add(podAllUsed, getPodMetricUsage(podMetric))
+		podMissingUsed = quotav1.Add(podMissingUsed, getPodMetricUsage(podMetric))
+		hasMissingPod = true
+		klog.V(6).InfoS("batch resource got missing pods", "node", node.Name, "pod", podKey)
+	}
+	if hasMissingPod {
```

Comment:
hasMissingPod seems not necessary since podMissingUsed will be zero if hasMissingPod is false

Reply:
this check was just for logging. now always log the dangling used with v=6